### DISCUSSION TITLE
Which class to use when you make a mixin for both family and non-family AutoDisposeAsyncNotifier<T>?

### DISCUSSION THREAD

**Discussion by MkamaMakoko** [rrousselGit/riverpod#2346] [Q&A] [answered]
I tried using `BuildlessAutoDisposeAsyncNotifier<T>` , it works but I got a warning for using it.

All my providers are generated, and I wanted to create a mixin for just modifying `state` and reusing some lines of code.

**Comment by rrousselGit** [comment, marked as answer]
You can do:

```dart
mixin Example<T> {
  Ref<T> get ref;
}
```

**Reply by MkamaMakoko** [reply]
So, something like
```
mixin Example<T>{
set state(AsyncValue<T> newState);
}
```
Is what I need.

Thanks.
